llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -31,6 +31,44 @@ using namespace llvm;
 
 namespace {
 
+struct VLInfo {
+  VLInfo(const MachineOperand &VLOp) {
+    IsImm = VLOp.isImm();
+    if (IsImm)
+      Imm = VLOp.getImm();
+    else
+      Reg = VLOp.getReg();
+  }
+
+  Register Reg;
```

Comment:
Is it possible to use `std::variant` here?

Reply:
updated to use std::variant.